tests/recipes/cpc2/test_predict.py

```diff
@@ -0,0 +1,71 @@
+"""Tests for the CPC2 predict functions."""
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from recipes.cpc2.baseline.predict import LogisticModel, make_disjoint_train_set
+
+
+# pylint: disable=redefined-outer-name
+@pytest.fixture
+def model():
+    """Return a LogisticModel instance."""
+    model = LogisticModel()
+    model.fit(np.array([0, 1, 2, 3, 4]), np.array([0, 25, 50, 75, 100]))
+    return model
+
+
+# pylint: disable=redefined-outer-name
+@pytest.mark.parametrize(
+    "model, value", [(model, 0.0), (model, 1.0), (model, 2.0)], indirect=["model"]
+)
+def test_logistic_model_symmetry(model: LogisticModel, value):
+    """Test the LogisticModel is symmetric."""
+    symmetric_value = 4 - value
+    assert model.predict(value) + model.predict(symmetric_value) == pytest.approx(
+        100.0, abs=1e-3
+    )
+
+
+# pylint: disable=redefined-outer-name
+@pytest.mark.parametrize(
+    "model, value",
+    [(model, -100.0), (model, -200.0), (model, 100.0), (model, 200.0)],
+    indirect=["model"],
```

Comment:
Oooh, not seen/used `indirect` before, thanks.

Reply:
Yes, it was new to me too. Took a bit of Googling. :smile: